File: backend/app/schemas/record.py
from datetime import datetime

from pydantic import BaseModel, Field


class RecordOut(BaseModel):
    """最高记录输出"""

    id: int
    academic_year_id: int
    event_name: str
    group_name: str
    gender: str
    sort_order: int
    holder_name: str
    result: str
    current_holder_name: str
    historical_result: str
    created_at: datetime

    class Config:
        from_attributes = True


class RecordCreate(BaseModel):
    """新增项目"""

    event_name: str = Field(max_length=64)
    group_name: str = Field(max_length=32)
    gender: str = Field(max_length=8)


class RecordUpdate(BaseModel):
    """更新记录（姓名/成绩/历史成绩/当前成绩创造者）"""

    holder_name: str = Field(default="", max_length=64)
    result: str = Field(default="", max_length=32)
    current_holder_name: str = Field(default="", max_length=64)
    historical_result: str = Field(default="", max_length=32)


class RecordSortUpdate(BaseModel):
    """批量更新排序"""

    updates: list[dict[str, int]]  # [{"id": 1, "sort_order": 0}, ...]

File: frontend/src/api/record.ts
import client from "./client";

export interface Record {
  id: number;
  academic_year_id: number;
  event_name: string;
  group_name: string;
  gender: string;
  sort_order: number;
  holder_name: string;
  result: string;
  current_holder_name: string;
  historical_result: string;
  created_at: string;
}

export interface RecordCreate {
  event_name: string;
  group_name: string;
  gender: string;
}

export interface RecordUpdate {
  holder_name: string;
  result: string;
  current_holder_name: string;
  historical_result: string;
}

export interface RecordSortUpdate {
  updates: Array<{ id: number; sort_order: number }>;
}

export async function getRecords(): Promise<Record[]> {
  const { data } = await client.get<Record[]>("/records");
  return data;
}

export async function createRecord(payload: RecordCreate): Promise<Record> {
  const { data } = await client.post<Record>("/records", payload);
  return data;
}

export async function updateRecord(id: number, payload: RecordUpdate): Promise<Record> {
  const { data } = await client.put<Record>(`/records/${id}`, payload);
  return data;
}

export async function updateRecordSort(payload: RecordSortUpdate): Promise<void> {
  await client.put("/records/sort", payload);
}

export async function deleteRecord(id: number): Promise<void> {
  await client.delete(`/records/${id}`);
}
